pvlib/test/test_irradiance.py

```diff
@@ -177,9 +177,11 @@ def test_perez_components():
         index=times
     )
     df_components = pd.DataFrame(components)
-    sum_components = pd.Series(np.sum(list(components.values()), axis=0), index=times)
+    sum_components = df_components.sum(axis=1)
+
+    columns = df_components.columns
     assert_series_equal(out, expected, check_less_precise=2)
-    assert_frame_equal(df_components, expected_components)
+    assert_frame_equal(df_components[columns], expected_components[columns])
```

Comment:
Specifying the columns in this way has the downside that you're not actually testing if the order is as expected. You can remove the [columns] and test the order if you change the way the dataframe is created. Two options for that:

```python
expected_components = pd.DataFrame(index=times)
expected_components['first_name'] = first_values
# and so on...

expected_components = pd.DataFrame(np.array([first_values, second_values, third_values]).T, 
    columns=['first', 'second', 'third'], index=times)
```

Not sure if you'd actually need the transpose there or not.